From 77de188451c43de5dfce6e6330752df2dbd8d6c2 Mon Sep 17 00:00:00 2001
From: Jiangli Yuan <jlyuan@motorola.com>
Date: Fri, 19 Feb 2016 12:33:47 -0800
Subject: [PATCH 685/959] IKSWM-19916:mmc: map cmdq DCMD task desc page as DMA
 mapping.

 The DCMD task desc page was mapped as normal memory pages,
 which will cause CMDQ ADMA fetches wrong datas, since
 the data may be still in our cpu caches. So remap the dcmd
 task desc page as DMA mapping.

   This fix is from QCOM.

Change-Id: I61a61e942eb34d29a8025563758d6dfa6d4349b6
Signed-off-by: Jiangli Yuan <jlyuan@motorola.com>
Reviewed-on: http://gerrit.mot.com/823744
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Zhi-Ming Yuan <a14194@motorola.com>
Reviewed-by: Sheng-Zhe Zhao <a18689@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/mmc/host/cmdq_hci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/cmdq_hci.c b/drivers/mmc/host/cmdq_hci.c
index fbb0117f96a..81062a7ca4d 100644
--- a/drivers/mmc/host/cmdq_hci.c
+++ b/drivers/mmc/host/cmdq_hci.c
@@ -591,7 +591,7 @@ static void cmdq_prep_dcmd_desc(struct mmc_host *mmc,
 		mrq->cmd->opcode, timing, resp_type);
 	dataddr = (__le64 __force *)(desc + 4);
 	dataddr[0] = cpu_to_le64((u64)mrq->cmd->arg);
-
+	dma_map_single(mmc_dev(mmc), (void *)task_desc, cq_host->task_desc_len, DMA_BIDIRECTIONAL);
 }
 
 static int cmdq_request(struct mmc_host *mmc, struct mmc_request *mrq)
-- 
2.11.0

